--- FIGlet.pm	Fri Dec  1 16:43:45 2000
+++ minifig.pl	Fri Dec  1 16:48:12 2000
@@ -117,11 +126,14 @@
 
 	($end) = /(.)\s*$/;
 	if( $wLead && /^(\s+)/ ){
-	    $wLead = length($1) < $wLead ? length($1) : $wLead; }
+	    $wLead = length($1) < $wLead ? length($1) : $wLead;
+	}
 	else{
-	    $wLead = 0; }
+	    $wLead = 0;
+	}
 	if( $wTrail && /([$self->{_header}->[0]\s]+)$end+\s*$/ ){
-	    $wTrail = length($1) < $wTrail ? length($1) : $wTrail; }
+	    $wTrail = length($1) < $wTrail ? length($1) : $wTrail;
+	}
 	else{
 	    $wTrail = 0; }
 	$length = $length > length($_) ? $length : length($_);
@@ -153,6 +165,13 @@
     $opts{-w} ||= 80;
 
     #Do text formatting here...
+    if( $opts{-X} ne 'L' ){
+	$opts{-X} ||= $self->{_header}->[6] ? 'R' : 'L';
+    }
+    if( $opts{-X} eq 'R' ){
+	$opts{-A} = join('', reverse(split('', $opts{-A})));
+    }
+
     $opts{-A} =~ tr/\t/ /;
     $opts{-A} =~  s%$/%\n%;
     if( $opts{-m} eq '-0' ){
@@ -171,6 +190,7 @@
     }
     @text = split("\n", $opts{-A});
 
+
     foreach( @text ){
 	s/^\s*//o;
 	my @lchars = map(ord($_), split('', $_));
@@ -186,10 +206,18 @@
 		if( $self->{-m} ne '-0' ){
 		    $line =~ s/$self->{_header}->[0]/ /g; }
 	    }
-	    if( $opts{-c} ){
-		$line = " "x(($opts{-w}-length($line))/2) . $line; }
-	    if( $opts{-r} ){
-		$line = " "x($opts{-w}-length($line)) . $line; }
+
+
+	    #Do some more text formatting here...
+	    if( $opts{-x} ne 'l' ){
+		$opts{-x} ||= $self->{-X} eq 'R' ? 'r' : 'l';
+	    }
+	    if( $opts{-x} eq 'c' ){
+		$line = " "x(($opts{-w}-length($line))/2) . $line;
+	    }
+	    if( $opts{-x} eq 'r' ){
+		$line = " "x($opts{-w}-length($line)) . $line;
+	    }
 	    $buffer .= "$line$/";
 	}
     }
